qa/workunits/windows/py_tests/internal/task_group.py

```diff
@@ -0,0 +1,62 @@
+from concurrent import futures
+import logging
+import threading
+
+
+LOG = logging.getLogger()
+
+
+class TaskGroup(object):
+    def __init__(self, max_workers=1, stop_on_error=True):
+        self._executor = futures.ThreadPoolExecutor(max_workers=max_workers)
+        self._lock = threading.Lock()
+
+        self.errors = 0
+        self.completed = 0
+        self.pending = 0
+
+        self.stopped = False
+        self.stop_on_error = stop_on_error
+
+        self._submitted_tasks = []
+
+    def _wrap_task(self, task):
+        def wrapper():
+            with self._lock:
+                if self.stopped:
+                    return
```

Comment:
The tasks are submitted in a tight loop but performed by a thread pool (4 threads by default), so it might be useful to exit early instead of waiting for all the tasks to complete. I can remove it though.

Reply:
I see -- we should see that with `--image-count=50` vs `--concurrency=8`.  In that case, it's probably useful, let's just get the counters right (although it's a minor nit).